AccountRepository.cls

```diff
@@ -0,0 +1,20 @@
+public class AccountRepository {
+    
+	public static Account findOneAccountByNameLike(String searchName) {
+        List<Account> accounts = [SELECT Id, Name FROM Account WHERE Name LIKE :('%' + searchName + '%') LIMIT 1];
```

Comment:
IF there are 2 accounts (1 - test1, 2- test2) here you don't know what the result, you just picked only 1 (assume it's test1)
Contact is related to test2 --> email won't be sent.